src/fivetran-destination/src/destination/ddl.rs

```diff
@@ -166,8 +179,65 @@ pub async fn handle_create_table(request: CreateTableRequest) -> Result<(), OpEr
     Ok(())
 }
 
-#[allow(clippy::unused_async)]
-pub async fn handle_alter_table(req: AlterTableRequest) -> Result<(), OpError> {
-    let error = format!("alter_table: {req:?}");
-    Err(OpErrorKind::Unsupported(error).into())
+pub async fn handle_alter_table(request: AlterTableRequest) -> Result<(), OpError> {
+    let (dbname, client) = config::connect(request.configuration).await?;
+
+    // Bail early if there isn't a table to alter.
+    let Some(request_table) = request.table else {
+        return Ok(());
+    };
+
+    let current_table = describe_table(&client, &dbname, &request.schema_name, &request_table.name)
+        .await
+        .context("alter table")?;
+    let Some(current_table) = current_table else {
+        return Err(OpErrorKind::UnknownTable {
+            database: dbname,
+            schema: request.schema_name,
+            table: request_table.name,
+        }
+        .into());
+    };
+
+    if columns_match(&request_table, &current_table) {
+        Ok(())
+    } else {
+        let error = format!("alter_table: {:?}", request_table);
+        Err(OpErrorKind::Unsupported(error).into())
+    }
+}
+
+// TODO(parkmycar): Implement some more complex diffing logic.
+fn columns_match(request: &Table, current: &Table) -> bool {
+    #[derive(Clone, Debug)]
+    struct ColumnMetadata {
+        ty: DataType,
+        primary_key: bool,
+        decimal: Option<DecimalParams>,
+    }
+
+    impl PartialEq<ColumnMetadata> for ColumnMetadata {
+        fn eq(&self, other: &ColumnMetadata) -> bool {
+            self.ty == other.ty
+                && self.primary_key == other.primary_key
+                // TODO(parkmycar): Better comparison for decimals.
+                && self.decimal.is_some() == other.decimal.is_some()
```

Comment:
How much do we need to worry about this not actually checking equality of the `decimal` field?

Reply:
Right now we don't need to worry much. I'm going to follow this up with another PR that will handle this though